docs/src/languages/php/xdebug.md

```diff
@@ -4,76 +4,95 @@ weight: 6
+When that key is defined, Platform.sh starts a second PHP-FPM process on the container that's identically configured but also has Xdebug enabled.
+Only incoming requests that have an Xdebug cookie or query parameter set are forwarded to the debug PHP-FPM process.
+All other requests are directed to the normal PHP-FPM process and thus have no performance impact.
 
-Xdebug has numerous other configuration options available.  They are all set as `php.ini` values, and can be configured the same way as any other [`php.ini` setting](/languages/php/ini.md).  Consult the [Xdebug documentation](https://xdebug.org/docs/) for a full list of available options, although in most cases the default configuration is sufficient.
+If you have the [router cache](../../define-routes/cache.md) enabled, you need to explicitly add the Xdebug cookie (`XDEBUG_SESSION`) to the cookie whitelist. Depending on the cookies you already have listed there the result should look similar to this:
 
-If you have the [router cache](../../define-routes/cache.md) enabled, you will also need to explicitly add the Xdebug cookie (`XDEBUG_SESSION`) to the cookie whitelist.  Depending on the cookies you already have listed there the result should look similar to this:
-
-```yaml
+```yaml {location=".platform/routes.yaml"}
 "https://{default}/":
     # ...
     cache:
         enabled: true
         cookies: ['/^SS?ESS/', 'XDEBUG_SESSION']
 ```
 
-## Using Xdebug
+Xdebug has numerous configuration options available.
+They can be set the same way as any other [PHP setting](./_index.md#php-settings).
+For a full list of available options, consult the [Xdebug documentation](https://xdebug.org/docs/).
+
+## 2. Use Xdebug
 
 ### Open a tunnel
 
-From your local checkout of your application, run `platform environment:xdebug` (or just `platform xdebug`) to open an SSH tunnel to the server.  That SSH tunnel will allow your IDE and the server to communicate debug information securely.
+From your local checkout of your application, run `platform environment:xdebug` (or just `platform xdebug`) to open an SSH tunnel to the server. That SSH tunnel allows your IDE and the server to communicate debug information securely.
 
-By default, Xdebug operates on port 9003. (Xdebug 2 used port 9000). Generally, it is best to configure your IDE to use that port.  If you wish to use an alternate port use the `--port` flag.
+By default, Xdebug operates on port 9003. Generally, it's best to configure your IDE to use that port. If you wish to use an alternate port use the `--port` flag.
 
 To close the tunnel and terminate the debug connection, press `Ctrl-C`.
 
 {{< note title="On Dedicated Generation 3" >}}
-Note that because you have several VMs running but your tunnel is connected to only one of them, your requests won't always reach the same host. 
+Note that because you have several VMs running, but your tunnel is connected to only one of them, your requests don't always reach the same host.
```

Comment:
:clipboard: 
```suggestion
Note that because you have several VMs running but your tunnel is connected to only one of them,
your requests don't always reach the same host.
```